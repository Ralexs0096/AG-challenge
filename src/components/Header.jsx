import { useState } from 'react'

const Header = ({ country, setCountry }) => {
  const [input, setInput] = useState('')
  const [alert, setAlert] = useState(false)

  const handleOnClick = () => {
    if (input.trim() === '') {
      setAlert(true)
      return
    }
    setAlert(false)
    setCountry(input)
    setInput('')
  }

  return (
    <div className="container container-header">
      <h1>Covid 19 Statistics</h1>
      <input
        type="text"
        className="input-search"
        name="input"
        value={input}
        placeholder="Search by Country"
        onChange={e => setInput(e.target.value)}
      />
      <button className="button-filter" onClick={handleOnClick}>
        Fetch
      </button>
      {alert ? (
        <div className="alert">Enter the name of any country</div>
      ) : null}
    </div>
  )
}
export default Header
