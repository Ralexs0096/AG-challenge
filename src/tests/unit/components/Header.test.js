import React from 'react'
import '@testing-library/jest-dom'
import Header from '../../../components/Header'
import { shallow } from 'enzyme'

describe('<Header /> tests', () => {
  const setCountry = jest.fn()
  let wrapper = shallow(<Header setCountry={setCountry} />)

  beforeEach(() => {
    wrapper = shallow(<Header setCountry={setCountry} />)
  })

  test('should display the <Header /> component', () => {
    expect(wrapper).toMatchSnapshot()
  })

  test('should verify that error is shown when fetch button is pressed without input value', () => {
    wrapper.find('button').simulate('click')

    const alertMessage = wrapper.find('.alert').text().trim()
    expect(alertMessage).toBe('Enter the name of any country')
  })

  test('should call the setCountry and clear the input', () => {
    const value = 'guatemala'
    wrapper.find('input').simulate('change', { target: { value } })
    wrapper.find('button').simulate('click')
    expect(setCountry).toHaveBeenCalled()
    expect(wrapper.find('input').prop('value')).toBe('')
  })

  test('should check input change', () => {
    const input = wrapper.find('input')
    const value = 'costa rica'
    input.simulate('change', { target: { value } })
    expect(wrapper.find('input').prop('value')).toBe(value)
  })
})
