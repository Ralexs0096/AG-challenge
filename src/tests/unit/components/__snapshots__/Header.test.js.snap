// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> tests should display the <Header /> component 1`] = `
<div
  className="container container-header"
>
  <h1>
    Covid 19 Statistics
  </h1>
  <input
    className="input-search"
    name="input"
    onChange={[Function]}
    placeholder="Search by Country"
    type="text"
    value=""
  />
  <button
    className="button-filter"
    onClick={[Function]}
  >
    Fetch
  </button>
</div>
`;
